src/Commands/NewCommand.ts

```diff
@@ -0,0 +1,100 @@
+import { ConversionContext } from "../Conversions/ConversionContext";
+import { Import } from "../Languages/Imports/Import";
+import { NewProperties, NewInstantiationSyntaxKind } from "../Languages/Properties/NewProperties";
+import { Command } from "./Command";
+import { LineResults } from "./LineResults";
+import { Parameter } from "./Parameters/Parameter";
+import { RepeatingParameters } from "./Parameters/RepeatingParameters";
+import { SingleParameter } from "./Parameters/SingleParameter";
+
+/**
+ * A command for instantiating an object of a given type.
+ */
+export class NewCommand extends Command {
+    /**
+     * Metadata on what object to instantiate.
+     */
+    protected newProperties: NewProperties;
+
+    /**
+     * Information on parameters this command takes in.
+     */
+    private static parameters: Parameter[] = [
+        new SingleParameter("typeName", "The type of the object to instantiate.", true),
+        new RepeatingParameters(
+            "Arguments to pass into the constructor",
+            [
+                new SingleParameter("argument", "Argument to pass into the constructor", false)
+            ])
+    ];
+
+    /**
+     * @returns Information on parameters this command takes in.
+     */
+    public getParameters(): Parameter[] {
+        return NewCommand.parameters;
+    }
+
+    /**
+     * Initializes a new instance of the Command class.
+     * 
+     * @param context   The driving context for converting the command.
+     */
+    constructor(context: ConversionContext) {
+        super(context);
+
+        this.newProperties = this.language.properties.newProp;
```

Comment:
Why bother cache this? Everywhere else, they're just referenced as members of `this.language`. I'd rather not take up the memory unnecessarily.